Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -100,6 +100,9 @@ extension OpenFeatureAPI {
                         continuation.resume()
                         holder.removeAll()
                     }
+                    if $0 == .error {
+                        return
```

Comment:
This block should be the same as the one for `.ready`: maybe we could even have a single `if` line with both cases. 
I think `return` is not enough for `setProviderAndAwait` itself to return, and calling `continuation.resume()` is needed for that